lib/endpoints/class-wp-rest-posts-controller.php

```diff
@@ -89,12 +84,17 @@ public function register_routes() {
 	 * @return WP_Error|WP_REST_Response
 	 */
 	public function get_items( $request ) {
-		$args = (array) $request->get_params();
-		$args['post_type'] = $this->post_type;
-		$args['paged'] = $args['page'];
-		$args['posts_per_page'] = $args['per_page'];
+		$args                   = array();
+		$args['post_type']      = $this->post_type;
+		$args['paged']          = $request['page'];
+		$args['posts_per_page'] = $request['per_page'];
```

Comment:
`filter` is only for WP_Query arguments. `page` and `per_page` are our new, consistent with the rest of the controllers, arguments.